Assets/Scripts/Switch.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Switch : MonoBehaviour
+{
+    [SerializeField]
+    [Tooltip("The color of the switch. 0 if yellow, 1 if blue, 2 if red, 3 if green.")]
+    [Range(0, 3)]
+    private int switchColor;
```

Comment:
why an int here? That doesn't seem intuitive. You could make an enum that lives in this class called "SwitchColor." That will result in a drop down when you serialize it in the Unity editor. Much more human readable.